cmd/logviewer/main.go

```diff
@@ -0,0 +1,261 @@
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/gorilla/websocket"
+	"github.com/urfave/cli"
+)
+
+const (
+	defaultLogPath = "logs"
+	ansi           = "[\u001B\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*(?:;[a-zA-Z\\d]*)*)?\u0007)|(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PRZcf-ntqry=><~]))"
+)
+
+var (
+	nodeHelpTemplate = `NAME:
+   {{.Name}} - {{.Usage}}
+USAGE:
+   {{.HelpName}} {{if .VisibleFlags}}[global options]{{end}}
+   {{if len .Authors}}
+AUTHOR:
+   {{range .Authors}}{{ . }}{{end}}
+   {{end}}{{if .Commands}}
+GLOBAL OPTIONS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+VERSION:
+   {{.Version}}
+   {{end}}
+`
+
+	// address defines a flag for setting the address and port on which the node will listen for connections
+	address = cli.StringFlag{
+		Name:  "address",
+		Usage: "Address and port number on which the application will try to connect to the elrond-go node",
+		Value: "127.0.0.1:8080",
+	}
+
+	// logLevelPatterns defines the logger levels and patterns
+	logLevelPatterns = cli.StringFlag{
+		Name:  "level",
+		Usage: "This flag specifies the logger levels and patterns",
+		Value: "*:" + logger.LogInfo.String(),
+	}
+
+	//logFile is used when the log output needs to be logged in a file
+	logFile = cli.BoolFlag{
+		Name:  "file",
+		Usage: "will automatically log into a file",
+	}
+
+	// testServer is used when a test web socket is need for testing purposes
+	testServer = cli.BoolFlag{
+		Name:  "test-server",
+		Usage: "starts a test server that will send log lines continuously. For development and debugging purposes only.",
+	}
+
+	ansiCleaner = regexp.MustCompile(ansi)
+)
+
+var log = logger.GetOrCreate("logviewer")
+var cliApp *cli.App
+var webSocket *websocket.Conn
+var manualStop chan struct{}
+var fileForLogs *os.File
+
+func main() {
+	initCliFlags()
+	manualStop = make(chan struct{})
+	log.Info("log viewer application started", "version", cliApp.Version)
+
+	cliApp.Action = func(c *cli.Context) error {
+		return startLogViewer(c)
+	}
+
+	err := cliApp.Run(os.Args)
+	if err != nil {
+		log.Error(err.Error())
+		os.Exit(1)
+	}
+}
+
+func initCliFlags() {
+	cliApp = cli.NewApp()
+	cli.AppHelpTemplate = nodeHelpTemplate
+	cliApp.Name = "Elrond Node CLI App"
+	cliApp.Version = fmt.Sprintf("%s/%s/%s-%s", "1.0.0", runtime.Version(), runtime.GOOS, runtime.GOARCH)
+	cliApp.Usage = "This is the entry point for starting a new Elrond node - the app will start after the genesis timestamp"
+	cliApp.Flags = []cli.Flag{
+		address,
+		logLevelPatterns,
+		logFile,
+		testServer,
+	}
+	cliApp.Authors = []cli.Author{
+		{
+			Name:  "The Elrond Team",
+			Email: "contact@elrond.com",
+		},
+	}
+}
+
+func startLogViewer(ctx *cli.Context) error {
+	var err error
+	logLevelPattern := ctx.GlobalString(logLevelPatterns.Name)
+	logLevels, _, err := logger.ParseLogLevelAndMatchingString(logLevelPattern)
+	if err != nil {
+		return err
+	}
+
+	if ctx.GlobalBool(testServer.Name) {
+		_ = testing.StartTestWebSocket("127.0.0.1:8080")
```

Comment:
If we can send the address through a flag, the testserver should also start at that provided address. If it's moved as a separate binary it should have the same configurable option

Reply:
Done when moved to a new binary file